tests/state_migration_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package tests
+
+import (
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/node"
+	"github.com/klaytn/klaytn/node/cn"
+	"github.com/klaytn/klaytn/storage/database"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+)
+
+// continues occurrence of state trie migration and node restart must success
+func TestMigration_ContinuesRestartAndMigration(t *testing.T) {
+	fullNode, node, validator, chainID, workspace, richAccount, _, _ := newSimpleBlockchain(t, 100)
+	defer os.RemoveAll(workspace)
+
+	numTxs := []int{100, 1000, 10000}
+	for i := 0; i < len(numTxs)*5; i++ {
+		numTx := numTxs[i%len(numTxs)]
+		t.Log("attempt", strconv.Itoa(i), "deployRandomTxs of", strconv.Itoa(numTx))
+		deployRandomTxs(t, node.TxPool(), chainID, richAccount, numTx)
+		time.Sleep(10 * time.Second) // wait until txpool is flushed
+
+		startMigration(t, node)
+		time.Sleep(1 * time.Second)
+		t.Log("migration state before restart", node.ChainDB().InMigration())
+		fullNode, node = restartNode(t, fullNode, node, workspace, validator)
+	}
```

Comment:
We need to check if the migration succeeds or not.

Reply:
review reflected :)